recipes/aws-libfabric/all/conanfile.py

```diff
@@ -0,0 +1,115 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.35.0"
+
+class LibfabricConan(ConanFile):
+    name = "aws-libfabric"
+    description = "AWS Libfabric"
+    topics = ("fabric", "communication", "framework", "service")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aws/libfabric"
+    license = "BSD-2-Clause", "GPL-2.0-or-later"
+    settings = "os", "arch", "compiler", "build_type"
+    _providers = ["gni", "psm", "psm2", "sockets", "rxm", "tcp", "udp", "usnic", "verbs", "bgq", "shm", "efa", "rxd", "mrail", "rstream", "perf", "hook_debug"]
+    options = {
+        **{ p: [True, False, "shared"] for p in _providers },
+        **{
+            "shared": [True, False],
+            "fPIC": [True, False],
+            "with_libnl": [True, False],
+            "with_bgq_progress": ["auto", "manual"],
+            "with_bgq_mr": ["basic", "scalable"]
+        }
+    }
+    default_options = {
+        **{ p: False for p in _providers if p not in ["efa", ] },
+        **{
+            "shared": False,
+            "fPIC": True,
+            "tcp": True,
+            "with_libnl": False,
+            "with_bgq_progress": "manual",
+            "with_bgq_mr": "basic"
+        }
+    }
+    build_requires = "autoconf/2.71", "automake/1.16.3", "libtool/2.4.6"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.options.efa == None:
```

Comment:
I don't completely understand how you handle `efa`.
It's added with possible values `[True, False, "shared"]` in `config_options`, but is not in `default_options`.
And here it is tested against `None`, which is an invalid option value.

I would handle this option as follows.
1. add `efa`: [True, False, "shared"]` to `options`
2. add `efa: False` to `default_options`
3. set default values in `config_options`:
    ```python
    if self.settings.os == "Linux":
        self.options.efa = True
    ```
4. override no option in `configure`

This way, the default option values depend on the os, and there is no chance that user options are overridden.